### PR TITLE
Refactor feature flags

### PR DESCRIPTION
**Pull request by btmorr**
## overview

Refactor feature flags to use persistent data instead of environment variables because of unpredictable/undesirable behavior in Resin, and a couple of trivial fixes

## changelog

- (refactor) Make feature flags use persistent data instead of environment variables
- (chore) Remove some dead code and update comment imports in robot

## review requests

Tested on 🌔 🌔  and demo'ed to @Kadee80

### CODE REVIEW

File: api/opentrons/config/feature_flags.py [hunk 1]
```diff
@@ -1,26 +1,57 @@
...
+
+def get_feature_flag(name: str) -> bool:
+    settings = get_all_feature_flags()
+    return bool(settings.get(name))
+
+
+def get_all_feature_flags() -> dict:
+    try:
+        if os.path.exists(SETTINGS_PATH):
+            fd = open(SETTINGS_PATH, 'r')
```
Comment (mcous) [inline api/opentrons/config/feature_flags.py]: Python noob questions: what's the difference between doing this vs. the `with open(path, 'r') as file:` notation I've seen elsewhere (notably: `scripts/bump_version.py`)

File: api/opentrons/server/endpoints/update.py [hunk 2]
```diff
@@ -124,3 +126,35 @@
...
     return res
+
+
+async def set_feature_flag(request):
+    """
+    Post body must include the keys 'key' and 'value'. The values of these two
+    entries will be set in the settings file as a key-value pair.
+    """
+    try:
+        req = await request.text()
```
Comment (mcous) [inline api/opentrons/server/endpoints/update.py]: As mentioned in #1100, `data = await request.json()` should work

File: api/tests/opentrons/server/test_update_endpoints.py [hunk 3]
```diff
@@ -52,3 +53,31 @@
...
+        ff, 'SETTINGS_PATH', os.path.join(tmpd.name, 'settings.json'))
+
+    log_init()
+
+    app = init(loop)
+    cli = await loop.create_task(test_client(app))
+
+    r0 = await cli.get('/settings')
+    r0body = await r0.text()
+    print('r0body: {}'.format(r0body))
```
Comment (mcous) [inline api/tests/opentrons/server/test_update_endpoints.py]: Did you mean to leave this print in?

Comment (btmorr) [reply api/tests/opentrons/server/test_update_endpoints.py]: Good catch. I'll rip it out when I rebase on a couple of PRs that are going in right now

File: api/opentrons/config/feature_flags.py (hunk 1 above)
Comment (btmorr) [reply api/opentrons/config/feature_flags.py]: The only difference is if you want to pass the file descriptor out of a closure, which you can't do with `with`. That was the case in a prior iteration on this branch, but isn't any more. I'll clean it up so we don't have any accidental fds left open.

**Pull request merged**